lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp

```diff
@@ -1488,11 +1488,24 @@ void HdVP2Mesh::_UpdateDrawItem(
             // Assign with the index to the dormant wireframe color by default.
             colorIndices.resize(instanceCount, 0);
 
+            //
+            VtIntArray delegateToVp2Map
+                = static_cast<HdVP2Instancer*>(instancer)->ComputeInstanceIdToVp2InstanceIdArray(
+                    id);
+
+            bool visibleInstance = false;
+
             // Assign with the index to the active selection highlight color.
             if (const auto state = drawScene.GetActiveSelectionState(id)) {
                 for (const auto& indexArray : state->instanceIndices) {
                     for (const auto index : indexArray) {
-                        colorIndices[index] = 1;
+                        if (index < delegateToVp2Map.size()) {
+                            int vp2InstanceIndex = delegateToVp2Map[index];
```

Comment:
We need to convert the scene delegate instance id into the Maya instance id to know which instance to draw.